synapse/models/files.py

```diff
@@ -200,7 +203,41 @@ def getModelDefs(self):
                 ('file:filepath', ('comp', {'fields': (('file', 'file:bytes'), ('path', 'file:path'))}), {
                     'doc': 'The fused knowledge of the association of a file:bytes node and a file:path.',
                 }),
-
+                ('file:mime:pe:section', ('comp', {'fields': (
+                        ('file', 'file:bytes'),
+                        ('name', 'str'),
+                        ('sha256', 'hash:sha256'),
+                    )}), {
+                    'doc': 'The fused knowledge a file:bytes node containing a pe section.',
+                }),
+# TODO get type/langid figured out
+#                ('file:mime:pe:resource', ('comp', {'fields': (
+#                        ('file', 'file:bytes'),
+#                        ('type', 'typecode'),
+#                        ('langid', 'langcode'),
+#                        ('resource', 'file:bytes'))}), {
+#                    'doc': 'The fused knowledge of a file:bytes node containing a pe resource.',
+#                }),
+                ('file:mime:pe:export', ('comp', {'fields': (
+                        ('file', 'file:bytes'),
+                        ('name', 'str'))}), {
+                    'doc': 'The fused knowledge of a file:bytes node containing a pe named export.',
+                }),
+                ('file:mime:pe:vsvers:keyval', ('comp', {'fields': (
+                        ('name', 'str'),
+                        ('value', 'str'))}), {
+                    'doc': 'A key value pair found in a PE vsversion info structure.',
+                }),
+                ('file:mime:pe:vsvers:info', ('comp', {'fields': (
+                        ('file', 'file:bytes'),
+                        ('keyval', 'file:mime:pe:vsvers:keyval'))}), {
+                    'doc': 'knowledge of a file:bytes node containing vsvers info.',
+                }),
+                ('file:string', ('comp', {'fields': (
```

Comment:
Would it be worth adding a field for the string's encoding to `file:string`?

`('encoding', 'str')`

Reply:
@erhlee-bird Would that record what the byte encoding of the string was inside of the file?

Reply:
That was the thought

Reply:
I guess this kind of gets to a question of "is this form supposed to represent a string whose raw form could be decoded by extracting bytes from a file and decoding them (utf8, ascii, esc)" or also include strings which may be decoded dynamically (Extracted from runtime / dynamic / emulation / etc)?  If so, this could be a weird field to require in a comp since that would require all tools to provide their encoding in how they got the strings; which isn't always the case.

Reply:
strings are strings.  bytes are bytes :stuck_out_tongue: